addon/mixins/select.js

```diff
@@ -44,7 +46,11 @@ export default Mixin.create({
 
     let path = this.get('_labelPath');
     if (path && isPresent(checkValue)) {
-      return (typeof checkValue.get === 'function' ? checkValue.get(path) : void 0) || checkValue[path]
+      if (typeof checkValue.get === 'function') {
+        return checkValue.get(path) // if Ember.Object
+      } else {
+        return checkValue[path] // if JS object
+      }
```

Comment:
I can at least assert that `(condition ? value : void 0) || fallback)` is equivalent to `condition ? value : fallback`, and given that, I decided to make it an `if/then` just because I haven't had enough coffee. It could become `return checkValue.get === 'function' ? checkValue.get(path) : checkValue[path];` if you want.

Oh yeah, and lemme add semicolons to the ends of those lines. Been doing too much CoffeeScript.